utils.py

```diff
@@ -279,3 +280,51 @@ def one_hot_and_flatten(array_2d_sequence, num_classes):
   x = onehot(array_2d_sequence, num_classes=num_classes)
   x = np.reshape(x, [x.shape[0], -1])  # flatten
   return x
+
+
+def get_mutation_pair_from_tensor_index(tensor_index: Iterable) -> Tuple[Tuple[int, int], Tuple[int, int]]:
+    """Returns a mutation pair corresponding to an epistasis tensor index.
+    Example usage:
+    >>> tensor_index = [1, 0, 0]
+    >>> seq_b = [0, 2, 0]
+    >>> ref_seq = [0, 0, 0]
+    >>> combine_seqs(seq_a, seq_b,  ref_seq)
+    [[1, 2, 0]]
+    Args:
+      tensor_index: 4-element tuple index into tensor of dimension LxLxAxA.
+
+    Returns:
+      2-tuple of mutations ((i, a), (j, b)), where i, j are positions and a, b are amino acids.
+    """
+    if len(tensor_index) != 4:
+        raise ValueError('Must be an index of a tensor of dimension LxLxAxA')
+    position_0 = tensor_index[0]
+    aa_0 = tensor_index[2]
+    position_1 = tensor_index[1]
+    aa_1 = tensor_index[3]
+    return ((position_0, aa_0), (position_1, aa_1))
+
+
+def get_top_n_4d_tensor_indexes(interaction_tensor: np.ndarray, top_n: int, lowest: bool = False) -> List[Tuple]:
+  """Returns the coordinates (i, j, a, b) for the `top_n` indexes of mutations with high effect.
+
+  Args:
+    interaction_tensor: LxLxAxA 4D tensor.
+    top_n: the number of interactions to return.
+    lowest: if True, return the coordinates with lowest effect.
+
+  Returns:
+    A list of (position, position, amino acid, amino acid) index tuples.
```

Comment:
this uses a different convention for pairs of mutations than get_mutation_pair_from_tensor_index, which may make it vulnerable to bugs.

Reply:
Yeah fair. I guess the idea is to pass the result of `get_top_n_4d_tensor_indexes` into `get_mutation_pair_from_tensor_index`. But we can just combine them into `get_top_n_mutation_pairs`